wurst/lib/Lodash.wurst

```diff
@@ -0,0 +1,250 @@
+    let iter = list.iterator()
+    for elem from iter
+        result = transform.call(elem, result)
+    iter.close()
+    return result
+
+public function foldl<Q, T>(Q startValue, BiFunction<T, Q, Q> transform) returns Function<LinkedList<T>, Q>
+    return list -> foldl(startValue, transform, list)
+
+public function LinkedList<T>.foldl<Q, T>(Q startValue, BiFunction<T, Q, Q> transform) returns Q
+    return foldl(startValue, transform, this)
+
+public function every<T>(Predicate<T> predicate, LinkedList<T> list) returns bool
+    let iter = list.iterator()
+    for elem from iter
+        if not predicate.call(elem)
+            return false
+    iter.close()
+    return true
+
+public function every<T>(Predicate<T> predicate) returns Predicate<LinkedList<T>>
+    return list -> every(predicate, list)
+
+public function LinkedList<T>.every<T>(Predicate<T> predicate) returns bool
+    return every(predicate, this)
+
+public function any<T>(Predicate<T> predicate, LinkedList<T> list) returns bool
+    let iter = list.iterator()
+    for elem from iter
+        if predicate.call(elem)
+            return true
+    iter.close()
+    return false
+
+public function any<T>(Predicate<T> predicate) returns Predicate<LinkedList<T>>
+    return list -> any(predicate, list)
+
+public function LinkedList<T>.any<T>(Predicate<T> predicate) returns bool
+    return any(predicate, this)
+
+public function keys<T, Q>(IterableMap<T, Q> map) returns LinkedList<T>
+    let keys = new LinkedList<T>()
+    let iter = map.iterator()
+    for key from iter
+        keys.add(key)
+    iter.close()
+    return keys
+
+public function IterableMap<T, Q>.keys<T, Q>() returns LinkedList<T>
+    return keys(this)
+
+public function values<T, Q>(IterableMap<T, Q> map) returns LinkedList<Q>
+    let values = new LinkedList<Q>()
+    let iter = map.iterator()
+    for key from iter
+        values.add(map.get(key))
+    iter.close()
+    return values
+
+public function IterableMap<T, Q>.values<T, Q>() returns LinkedList<Q>
+    return values(this)
+
+public function map<T, Q>(Function<T, Q> transform, LinkedList<T> list) returns LinkedList<Q>
+    let result = new LinkedList<Q>()
+    let iter = list.iterator()
+    for elem from iter
+        result.add(transform.call(elem))
+    iter.close()
+    return result
+
+public function map<T, Q>(Function<T, Q> transform) returns Function<LinkedList<T>, LinkedList<Q>>
+    return list -> map(transform, list)
+
+public function flatten<T>(LinkedList<LinkedList<T>> list) returns LinkedList<T>
+    let result = new LinkedList<T>()
+    let iter = list.iterator()
+    for elem from iter
+        let subIter = elem.iterator()
+        for subElem from subIter
+            result.add(subElem)
+    iter.close()
+    return result
+
+public function flatten<T>() returns Function<LinkedList<LinkedList<T>>, LinkedList<T>>
+    return list -> flatten(list)
+
+public function drop<T>(int numElems, LinkedList<T> list) returns LinkedList<T>
```

Comment:
Dropping from the end with a negative argument would be nice. I think a generic `slice` makes the most sense and then this and `take` could just be calls to that.